src/modeler/AutoRest.Swagger/Validation/Extensions/LongRunningResponseValidationDelete.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Validation;
+using AutoRest.Core.Properties;
+using System;
+using System.Collections.Generic;
+using AutoRest.Swagger.Model;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class LongRunningResponseValidationDelete : LongRunningExtensionRule
+    {
+        /// <summary>
+        /// The template message for this Rule.
+        /// </summary>
+        public override string MessageTemplate => Resources.LongRunningResponsePostNotValid;
+
+        /// <summary>
+        /// An x-ms-long-running-operation extension passes this rule if the operation that this extension has a valid response defined.
+        /// </summary>
+        /// <param name="longRunning">long running extension object.</param>
+        /// <param name="context">Rule context.</param>
+        /// <param name="formatParameters">list of parameters for message formatter.</param>
+        /// <returns><c>true</c> if operation has valid response code modeled, otherwise <c>false</c>.</returns>
+        /// <remarks>This rule corresponds to M2005.</remarks>
+        public override bool IsValid(object longRunning, RuleContext context, out object[] formatParameters)
```

Comment:
I am applying this rule at extension level. I agree with you on Linq query but specifically for this case, I have concerns in moving them to the paths level:

If i apply it to the path level then I think we'll introduce the overhead of finding out all operations that has `x-ms-long-running` extension i.e. Linq has to search through all the operation to find subset. Secondly, as you are already working for some optimization around performance to speedup the validation process this would be like just opposite in the sense to incuring more cost.  

Instead I'd prefer to keep it as deep as possible whenever we can. What do you think about this?